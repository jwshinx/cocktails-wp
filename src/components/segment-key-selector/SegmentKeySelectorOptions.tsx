import React from 'react'

import { SegmentKeyType } from '../../models/cocktail'

const SegmentKeySelectorOptions = ({
  items,
}: {
  items: Array<SegmentKeyType>
}): JSX.Element => {
  return (
    <>
      {items.map((pair): JSX.Element => {
        return (
          <option key={pair.name} value={pair.name}>
            {pair.label}
          </option>
        )
      })}
    </>
  )
}

export default SegmentKeySelectorOptions
