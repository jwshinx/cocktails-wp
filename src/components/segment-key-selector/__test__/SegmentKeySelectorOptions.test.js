/**
 * @jest-environment jsdom
 */

import React from 'react'
// import ReactDOM from 'react-dom'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

import SegmentKeySelectorOptions from '../SegmentKeySelectorOptions'

const items = [
  { label: 'First Letter', name: 'firstLetter', value: 'f' },
  { label: 'Cocktail Name', name: 'cocktailName', value: 's' },
]

it('renders a "First Letter" option', () => {
  render(<SegmentKeySelectorOptions items={items} />)
  const elem = screen.getByText(/First Letter/i)
  expect(elem).toBeInTheDocument()
})

it('renders a "Cocktail Name" option', () => {
  render(<SegmentKeySelectorOptions items={items} />)
  const elem = screen.getByText(/Cocktail Name/i)
  expect(elem).toBeInTheDocument()
})

it('should snapshot SegmentKeySelectorOptions', () => {
  const tree = renderer
    .create(<SegmentKeySelectorOptions items={items} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
