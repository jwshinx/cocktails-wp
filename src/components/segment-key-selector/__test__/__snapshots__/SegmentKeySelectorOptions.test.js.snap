// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot SegmentKeySelectorOptions 1`] = `
Array [
  <option
    value="firstLetter"
  >
    First Letter
  </option>,
  <option
    value="cocktailName"
  >
    Cocktail Name
  </option>,
]
`;
